src/exchange/disclaimer/disclaimer.controller.js

```diff
@@ -28,44 +25,73 @@ angular
-                    this.disclaimersList = disclaimers;
-                    this.setMessagesFlags(disclaimers);
+                    for (let i = 0; i < disclaimers.list.results.length; i++) {
+                        this.disclaimersList.splice(i, 1, disclaimers.list.results[i]);
+                    }
+                    for (let i = disclaimers.list.results.length; i < this.disclaimersList.length; i++) {
+                        this.disclaimersList.splice(i, 1);
+                    }
+                    this.setMessagesFlags(this.disclaimers);
                 })
                 .catch((err) => {
                     this.services.messaging.writeError(this.services.translator.tr("exchange_tab_DISCLAIMER_error_message"), err);
-                })
-                .finally(() => {
-                    this.loading.table = false;
                 });
         }
 
-        setMessagesFlags (disclaimersList) {
-            const disclaimersHaveEmptylot = this.hasEmptySlot(disclaimersList.list.results);
-            const disclaimersHaveFullSlot = this.hasFullSlot(disclaimersList.list.results);
 
-            this.addDomainMessageFlag = _.isEmpty(disclaimersList.list.results) || (!disclaimersHaveEmptylot && !disclaimersHaveFullSlot);
-            this.noDisclaimerMessageFlag = disclaimersHaveEmptylot && !disclaimersHaveFullSlot;
+        loadPaginated ($config) {
+            this.disclaimersList = null;
+            this.pageSize = $config.pageSize;
+            this.offset = $config.offset;
+            return this._getDisclaimers($config)
+                .then((disclaimers) => {
+                    this.disclaimersList = disclaimers.list.results;
+                    this.setMessagesFlags(this.disclaimers);
+                    return {
+                        data: this.disclaimersList,
+                        meta: {
+                            totalCount: disclaimers.count
+                        }
+                    };
+                })
+                .catch((err) => {
+                    this.services.messaging.writeError(this.services.translator.tr("exchange_tab_DISCLAIMER_error_message"), err);
+                });
+        }
+
+        setMessagesFlags () {
+            this.addDomainMessageFlag = !this.newDisclaimersDisabled();
         }
 
         newDisclaimersDisabled () {
-            return this.disclaimersList != null && !this.hasEmptySlot(this.disclaimersList.list.results);
+            return !this.disclaimersList || !this.hasEmptySlot(this.disclaimersList);
         }
 
         addDisclaimer () {
             if (!this.newDisclaimersDisabled()) {
                 this.services.navigation.setAction("exchange/disclaimer/add/disclaimer-add");
             }
         }
+
+        updateDisclaimer (disclaimer) {
+            if (!disclaimer.taskPendingId) {
+                this.services.navigation.setAction("exchange/disclaimer/update/disclaimer-update", angular.copy(disclaimer));
```

Comment:
angular.copy(disclaimer) ... mmmh I already saw that forgotten things somewhere :P